processor/src/host/event_handling.rs

```diff
@@ -0,0 +1,39 @@
+use alloc::boxed::Box;
+use std::collections::BTreeMap;
+
+use crate::{ExecutionError, ProcessState};
+
+pub trait EventHandler<A> {
+    fn id(&self) -> u32;
+
+    // TODO(plafer): not sure if returning `ExecutionError` is the best choice
+    fn on_event(
+        &mut self,
+        process: ProcessState,
+        advice_provider: &mut A,
+    ) -> Result<(), ExecutionError>;
+}
+
+#[derive(Default)]
+pub struct EventHandlerRegistry<A> {
+    handlers: BTreeMap<u32, Box<dyn EventHandler<A>>>,
+}
+
+impl<A> EventHandlerRegistry<A> {
+    pub fn register_event_handler(&mut self, handler: Box<dyn EventHandler<A>>) {
+        self.handlers.insert(handler.id(), handler);
+    }
```

Comment:
Should we do something if a handler for this event has already been registered? Maybe return an error? Or maybe a boolean value saying that something was already registered?

Reply:
Good idea. I think an `Error` is the way to go (to avoid [boolean blindness](https://runtimeverification.com/blog/code-smell-boolean-blindness))